.github/workflows/validate.yml

```diff
@@ -19,20 +19,58 @@ env:
   BUILDX_VERSION: "v0.9.1"  # leave empty to use the one available on GitHub virtual environment
 
 jobs:
+  list-commits:
+    runs-on: ubuntu-20.04
+    outputs:
+      commits: ${{ steps.commits.outputs.matrix }}
+    steps:
+      -
+        name: List commits
+        id: commits
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let matrix = [];
+            core.info(`context.eventName: ${context.eventName}`);
+            if (context.eventName != 'pull_request') {
+              core.info('Not a pull request, skipping');
+              matrix.push('');
+            } else {
+              await core.group(`Retrieving commits for PR ${context.issue.number}`, async () => {
+                const commits = await github.paginate(github.rest.pulls.listCommits.endpoint.merge({
+                  ...context.repo,
+                  pull_number: context.issue.number,
+                }));
+                for (const { sha } of commits) {
```

Comment:
Can we set a max number of commits as mentioned in https://github.com/moby/buildkit/issues/3077#issuecomment-1237302064?

Realistically it's not going to be abused that much (since pushing commits to the PR is the same permissions scope as modifying the github actions anyways), but it would be good to protect against PRs that accidentally have 100s of commits, e.g. when targeting the wrong branch for a PR , definitely never done that before :sweat_smile: 

IMO, we could just fail if the number of commits is larger than X, where X is some reasonably high number? Not quite sure what we should pick.

Reply:
How much is the real question here :sweat_smile:

Reply:
Sorry for being so long on this - I really want to get this in, but I can't think of a reasonable amount here :scream: 

Ideally, whatever the cap is, we (as maintainers) should be able to override it and run tests over all the commits - maybe using some slash commands, or maybe we could use a `workflow_dispatch` trigger to manually disable the cap?

Anyways, for now, I think a manual limit of "only build+validate the last 10 commits", but output some warning if there are untested commits would be good enough?

Reply:
Added a reusable workflow: https://github.com/crazy-max/.github/pull/5

Current limit `20`. Let me know if that sounds good.

Reply:
Yup, that sounds good to me.